lms/djangoapps/certificates/management/commands/regenerate_noidv_cert.py

```diff
@@ -88,10 +117,37 @@ def _handle_course(course_key, batch_size, sleep_seconds, count):
 
     log.info(f'Regenerating {len(certs)} unverified certificates for {course_key}')
 
+    return _regenerate_certs(certs, batch_size, sleep_seconds, count, False)
+
+
+def _handle_all_courses(batch_size, sleep_seconds, excluded_keys):
+    """
+    Regenerates unverified status certificates for all courses except for those designated by the excluded keys with
+    delay seconds between certs. Returns how many certs were originally unverified and regenerated.
+    """
+    certs = GeneratedCertificate.objects\
+        .filter(status=CertificateStatuses.unverified)\
+        .exclude(course_id__in=excluded_keys)
+    log.info(f'Regenerating {len(certs)} unverified certificates in all courses except for excluded keys')
+
+    return _regenerate_certs(certs, batch_size, sleep_seconds, 0, True)
+
+
+def _regenerate_certs(certs, batch_size, sleep_seconds, count, check_integrity_signature_flag):
+    """
+    Triggers generate certificate task for a given set of certificates
+    """
     for cert in certs:
-        user = User.objects.get(id=cert.user_id)
-        generate_certificate_task(user, course_key, generation_mode='batch', delay_seconds=0)
-        count += 1
-        if count % batch_size == 0:
-            time.sleep(sleep_seconds)
+        # not ideal to replicate this check, but we have to in the case that we are regenerating certs for all courses.
+        if check_integrity_signature_flag and not is_integrity_signature_enabled(cert.course_id):
```

Comment:
mentioned it a comment as well, but just wanted to point to the fact that this check exists twice in the command. I added the `check_integrity_signature_flag` to this to ensure that when we specify a course key, we don't waste time calling this for each cert. Ideally when we are running this command for all courses, we would be able to filter out courses that don't have this flag enabled before entering this loop, but I don't think it's possible to add that kind of logic to a django filter. In _handle_all_courses I could filter through the list of certs before they reach this function, but it would be inefficient to loop through the list of certs twice (once to get rid of certs in courses where the flag is not enabled, and one to regenerate those certs).